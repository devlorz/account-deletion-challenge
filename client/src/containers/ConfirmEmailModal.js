import _ from 'lodash'
import PropTypes from 'prop-types'
import React from 'react'

import { isLoading } from '../LoadState'
import FormInputPassword from '../components/FormInputPassword'

class ConfirmEmailModal extends React.PureComponent {
  static propTypes = {
    onClickToDelete: PropTypes.func,
    onBackButton: PropTypes.func,
    email: PropTypes.string,
    onTypeEmail: PropTypes.func,
    resetTerminateAccountStatus: PropTypes.func,
    terminateAccountStatus: PropTypes.object,
  }

  state = { markedConsequences: false }

  componentWillUnmount() {
    this.props.resetTerminateAccountStatus()
  }

  getStateButton = () => {
    if (isLoading(this.props.terminateAccountStatus)) return true
    if (this.state.markedConsequences && this.props.email) return false
    return true
  }

  onToggleMarkedConsequences = () => {
    this.setState({ markedConsequences: !this.state.markedConsequences })
  }

  renderFormInputPasssword = () => {
    const errorMessage = _.get(this.props.terminateAccountStatus, 'error', null)
    return (
      <FormInputPassword
        onTypeEmail={this.props.onTypeEmail}
        errorMessage={errorMessage}
      />
    )
  }

  render() {
    return (
      <div>
        <h1>Delete account</h1>
        <p>This action cannot be undone.</p>
        <div>Please enter your email: {this.renderFormInputPasssword()}</div>
        <div style={{ marginTop: '1rem' }}>
          <label>
            <input
              type="checkbox"
              checked={this.state.markedConsequences}
              onChange={this.onToggleMarkedConsequences}
            />
            I understand the consequences.
          </label>
        </div>
        <div>
          <button onClick={this.props.onBackButton}>Back</button>
          <button
            onClick={this.props.onClickToDelete}
            disabled={this.getStateButton()}
          >
            Delete my account
          </button>
        </div>
      </div>
    )
  }
}

export default ConfirmEmailModal
