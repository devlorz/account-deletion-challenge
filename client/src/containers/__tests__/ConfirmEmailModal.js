import React from 'react'
import { render } from 'react-testing-library'

import ConfirmEmailModal from '../ConfirmEmailModal'

test('snapshot', () => {
  const { container } = render(
    <ConfirmEmailModal
      onClickToDelete={jest.fn()}
      onBackButton={jest.fn()}
      email={'test@mail.com'}
      onTypeEmail={jest.fn()}
      terminateAccountStatus={{}}
      resetTerminateAccountStatus={jest.fn()}
    />
  )
  expect(container.firstChild).toMatchSnapshot()
})
