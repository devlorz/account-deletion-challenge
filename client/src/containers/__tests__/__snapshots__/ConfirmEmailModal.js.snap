// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  data-reactroot=""
>
  <h1>
    Delete account
  </h1>
  <p>
    This action cannot be undone.
  </p>
  <div>
    <!-- react-text: 5 -->
    Please enter your email: 
    <!-- /react-text -->
    <div>
      <input
        placeholder="ross@example.com"
        style="width: 350px;"
        type="text"
      />
      <span
        style="color: red;"
      />
    </div>
  </div>
  <div
    style="margin-top: 1rem;"
  >
    <label>
      <input
        type="checkbox"
        value="on"
      />
      <!-- react-text: 12 -->
      I understand the consequences.
      <!-- /react-text -->
    </label>
  </div>
  <div>
    <button>
      Back
    </button>
    <button
      disabled=""
    >
      Delete my account
    </button>
  </div>
</div>
`;
